mlir/lib/Transforms/Utils/DialectConversion.cpp

```diff
@@ -970,6 +971,54 @@ class BlockTypeConversionAction : public BlockAction {
 
   void rollback() override;
 };
+
+/// An operation rewrite.
```

Comment:
Can you expand on the role of the class, the context where it's used?

The "Action" name for this whole section is not great by the way, since the concept of "Actions" is now core to MLIR tracing...

Reply:
I renamed the class to `OperationRewrite`. I added more documentation to the superclass `IRRewrite` in the dependent PR.